frontend/src/hooks/queryBuilder/useFetchKeysAndValues.ts

```diff
@@ -54,49 +76,56 @@ export const useFetchKeysAndValues = (
 	const handleFetchOption = async (
 		value: string,
 		query: IBuilderQuery,
+		keys: BaseAutocompleteData[],
 	): Promise<void> => {
-		if (value) {
-			// separate the search value into the attribute key and the operator
-			const [tKey, operator] = separateSearchValue(value);
-			setResults([]);
-			if (tKey && operator) {
-				const { payload } = await getAttributesValues({
-					searchText: searchValue,
-					dataSource: query.dataSource,
-					aggregateOperator: query.aggregateOperator,
-					aggregateAttribute: query.aggregateAttribute.key,
-					attributeKey: tKey,
-				});
-				if (payload) {
-					const values = Object.values(payload).find((el) => !!el);
-					if (values) {
-						setResults(values);
-					} else {
-						setResults([]);
-					}
-				}
-			}
+		if (!value) {
+			return;
+		}
+		const { tagKey, tagOperator, tagValue } = getTagToken(value);
+		const filterAttributeKey = keys.find((key) => tagKey.includes(key.key));
```

Comment:
Why is this `tagKey.includes(key.key)`?

Say I have keys as
```javascript
keys = [
    {
        "key": "http_status",
        "dataType": "number",
        "type": "tag",
        "isColumn": false
    },
    {
        "key": "http_status_text",
        "dataType": "string",
        "type": "tag",
        "isColumn": false
    }
]

```

and tagKey = "http_status_text", why would this send `http_status` as `attributeKey` and `number` as `filterAttributeKeyDataType ` when it should be sending `http_status_text` and `string`?

Reply:
@srikanthccv 
https://github.com/SigNoz/signoz/pull/2629/commits/879051d28b579927c8087bbda08584327655ae7f